lib/Doctrine/Persistence/Mapping/Driver/AnnotationDriver.php

```diff
@@ -220,7 +220,7 @@ public function getAllClassNames()
             foreach ($iterator as $file) {
                 $sourceFile = $file[0];
 
-                if (! preg_match('(^phar:)i', $sourceFile)) {
+                if (preg_match('(^phar:)i', $sourceFile) === 0) {
```

Comment:
#50 wrongly uses `=== false` here. `false` is returned if an error occurs, the correct to way to assert that there is no match is to count them and assert that that count is 0.